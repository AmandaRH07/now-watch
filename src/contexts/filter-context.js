import React, { createContext, useContext, useState } from 'react';

const FilterContext = createContext()
const FilterProvider = ({ children }) => {
  const [checkboxService, setCheckboxService] = useState({
    apple: false,
    prime: false,
    disney: false,
    hbo: false,
    netflix: false,
    starz: false
  });

  const [checkboxType, setCheckboxType] = useState({
    movie: false,
    series: false,
  });

  const [checkboxGenre, setCheckboxGenre] = useState({
    Acao28: false,
    Animacao16: false,
    Biografia1: false,
    Comedia38: false,
    Documentario99: false,
    Drama18: false,
    FiccaoCientifica878: false,
    Romance10749: false,
    Terror27: false
  });

  const [filterService, setFilterService] = useState([]);
  const [filterType, setFilterType] = useState([]);
  const [filterGenre, setFilterGenre] = useState([]);
  const [responseHasMore, setResponseHasMore] = useState();
  const [responseNextCursor, setResponseNextCursor] = useState();

  const [responseData, setResponseData] = useState([]);
  return (
    <FilterContext.Provider
      value={{
        checkboxService,
        setCheckboxService,
        checkboxType,
        setCheckboxType,
        checkboxGenre,
        setCheckboxGenre,
        filterService,
        setFilterService,
        filterType,
        setFilterType,
        filterGenre,
        setFilterGenre,
        responseHasMore,
        setResponseHasMore,
        responseNextCursor,
        setResponseNextCursor,
        responseData,
        setResponseData
      }}>
      {children}
    </FilterContext.Provider>
  )
}

export const useFilterContext = () => useContext(FilterContext);
export { FilterProvider, FilterContext }