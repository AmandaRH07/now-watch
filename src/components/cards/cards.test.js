import Cards from "."
import { render } from "@testing-library/react"
import { FilterProvider } from "../../contexts/filter-context"

const renderComponent = () => (
  render(
    <FilterProvider>
    <Cards />
    </FilterProvider>
  )
)

describe('<Cards/>', () => {

  it('Should render Cards', () => {
    const {container} = renderComponent();

    console.log("container", container);
    expect(container).toMatchSnapshot();
  })
})