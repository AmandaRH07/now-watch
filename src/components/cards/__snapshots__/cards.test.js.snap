// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards/> Should render Cards 1`] = `
<div>
  <div
    class="cards-conteiner"
  >
    <div
      class="card-content"
    />
    <div
      class="footer-content"
    />
  </div>
</div>
`;
